Sources/Subprocess/Platforms/Subprocess+Darwin.swift

```diff
@@ -144,184 +144,223 @@ extension Configuration {
-        } else if spawnAttributeError == 0 && self.platformOptions.qualityOfService == .background {
-            spawnAttributeError = posix_spawnattr_set_qos_class_np(&spawnAttributes, QOS_CLASS_BACKGROUND)
-        }
 
-        // Setup cwd
-        var chdirError: Int32 = 0
-        if intendedWorkingDir != .currentWorkingDirectory {
-            chdirError = intendedWorkingDir.withPlatformString { workDir in
+            // Setup cwd
+            let intendedWorkingDir = self.workingDirectory.string
+            let chdirError: Int32 = intendedWorkingDir.withPlatformString { workDir in
                 return posix_spawn_file_actions_addchdir_np(&fileActions, workDir)
             }
-        }
 
-        // Error handling
-        if chdirError != 0 || spawnAttributeError != 0 {
-            try self.cleanupPreSpawn(input: inputPipe, output: outputPipe, error: errorPipe)
-            if spawnAttributeError != 0 {
-                throw SubprocessError(
-                    code: .init(.spawnFailed),
-                    underlyingError: .init(rawValue: spawnAttributeError)
-                )
+            // Error handling
+            if chdirError != 0 || spawnAttributeError != 0 {
+                try self.cleanupPreSpawn(input: inputPipe, output: outputPipe, error: errorPipe)
+                if spawnAttributeError != 0 {
+                    throw SubprocessError(
+                        code: .init(.spawnFailed),
+                        underlyingError: .init(rawValue: spawnAttributeError)
+                    )
+                }
+
+                if chdirError != 0 {
+                    throw SubprocessError(
+                        code: .init(.spawnFailed),
+                        underlyingError: .init(rawValue: spawnAttributeError)
+                    )
+                }
+            }
+            // Run additional config
+            if let spawnConfig = self.platformOptions.preSpawnProcessConfigurator {
+                try spawnConfig(&spawnAttributes, &fileActions)
             }
 
-            if chdirError != 0 {
+            // Spawn
+            let spawnError: CInt = possibleExecutablePath.withCString { exePath in
+                return supplementaryGroups.withOptionalUnsafeBufferPointer { sgroups in
+                    return _subprocess_spawn(
+                        &pid,
+                        exePath,
+                        &fileActions,
+                        &spawnAttributes,
+                        argv,
+                        env,
+                        uidPtr,
+                        gidPtr,
+                        Int32(supplementaryGroups?.count ?? 0),
+                        sgroups?.baseAddress,
+                        self.platformOptions.createSession ? 1 : 0
+                    )
+                }
+            }
+            // Spawn error
+            if spawnError != 0 {
+                if spawnError == ENOENT {
```

Comment:
@iCharlesHu What if the `ENOENT` refers to the current working directory and _not_ the executable?